2024Robot_Team3484/src/main/cpp/subsystems/LauncherSubsystem.cpp

```diff
@@ -24,40 +24,40 @@ LauncherSubsystem::LauncherSubsystem(
     Launcher_Encoder_Left = new SparkRelativeEncoder(_left_motor.GetEncoder(rev::SparkRelativeEncoder::Type::kQuadrature, 4096 ));
     Launcher_Encoder_Right = new SparkRelativeEncoder(_right_motor.GetEncoder(rev::SparkRelativeEncoder::Type::kQuadrature, 4096 ));
 
-    Launcher_m_Left_pidController = _left_motor.GetPIDController();
-    Launcher_m_Right_pidController = _right_motor.GetPIDController();
+    Launcher_m_Left_pidController = new SparkPIDController(_left_motor.GetPIDController());
+    Launcher_m_Right_pidController = new SparkPIDController(_right_motor.GetPIDController());
 
-    Launcher_m_Left_pidController.SetFeedbackDevice(*Launcher_Encoder_Left);
-    Launcher_m_Right_pidController.SetFeedbackDevice(*Launcher_Encoder_Right);
+    Launcher_m_Left_pidController->SetFeedbackDevice(*Launcher_Encoder_Left);
+    Launcher_m_Right_pidController->SetFeedbackDevice(*Launcher_Encoder_Right);
 
     _left_motor.RestoreFactoryDefaults();
     _right_motor.RestoreFactoryDefaults();
 
-    _left_motor.SetInverted(MOTER_INVERTED);
-    _right_motor.SetInverted(!MOTER_INVERTED);
+    _left_motor.SetInverted(MOTOR_INVERTED);
+    _right_motor.SetInverted(!MOTOR_INVERTED);
 
-    Launcher_m_Left_pidController.SetP(P_Launcher);
-    Launcher_m_Left_pidController.SetI(I_Launcher);
-    Launcher_m_Left_pidController.SetD(D_Launcher);
-    Launcher_m_Left_pidController.SetIZone(0);
-    Launcher_m_Left_pidController.SetFF(FF_Launcher);
-    Launcher_m_Left_pidController.SetOutputRange(0, 1);
+    Launcher_m_Left_pidController->SetP(P_Launcher);
+    Launcher_m_Left_pidController->SetI(I_Launcher);
+    Launcher_m_Left_pidController->SetD(D_Launcher);
+    Launcher_m_Left_pidController->SetIZone(0);
+    Launcher_m_Left_pidController->SetFF(FF_Launcher);
+    Launcher_m_Left_pidController->SetOutputRange(0, 1);
 
-    Launcher_m_Right_pidController.SetP(P_Launcher);
-    Launcher_m_Right_pidController.SetD(D_Launcher);
-    Launcher_m_Right_pidController.SetI(I_Launcher);
-    Launcher_m_Right_pidController.SetIZone(0);
-    Launcher_m_Right_pidController.SetFF(FF_Launcher);
-    Launcher_m_Right_pidController.SetOutputRange(0, 1);
+    Launcher_m_Right_pidController->SetP(P_Launcher);
+    Launcher_m_Right_pidController->SetD(D_Launcher);
+    Launcher_m_Right_pidController->SetI(I_Launcher);
+    Launcher_m_Right_pidController->SetIZone(0);
+    Launcher_m_Right_pidController->SetFF(FF_Launcher);
+    Launcher_m_Right_pidController->SetOutputRange(0, 1);
```

Comment:
Since I recommended this change on the intake and drivetrain subsystems, I suggest using `SC::SC_PIDConstants` in place of `int P_Launcher, int I_Launcher, int D_Launcher, int FF_Launcher` in the constructor's parameter list. i.e.:
```c++
LauncherSubsystem::LauncherSubsytem(int CAN_ID_Left, int CAN_ID_Right, SC::SC_PIDConstants pidc, double RPM_InRangeWindow)
{
    // Create motor instances
    // Create encoder instances
    // Create PID Controller instances

    // Check launcher PID controllers aren't null
    Launcher_PID_Left->SetP(pidc.Kp);
    Launcher_PID_Left->SetI(pidc.Ki);
    Launcher_PID_Left->SetD(pidc.Kd);
    Launcher_PID_Left->SetFF(pidc.Kf);

   // Do the same thing here for the right side PID controller.
}
```